### DISCUSSION TITLE
Support for `leadingComments`, `innerComments` and `trailingComments` in AST nodes

### DISCUSSION THREAD

**Discussion by Dunqing** [oxc-project/oxc#7153] [Feature Request]
These properties are supported in `Babel`.

**Comment by Boshen** [comment]
This'll hurt performance, we'll have to look for better ways to support this.

**Comment by lukeed** [comment]
Bump on this

If you're looking to match babel output now, this is a must. To align `tree.comments` back with the node they were for, I have to parse twice or once w/ a cache of start positions. RE: performance, this is certainly worse than the minimal steps oxc needs to do during its single pass.
